### PR TITLE
Increase verbosity of Artemis provisioning errors

### PR DESCRIPTION
**Pull request by happz**
Some errors might have added context that would be helpful for investigation: the plugin itself is "just" an API client, and often the important info comes back in API responses.

Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]

File: tmt/steps/provision/artemis.py [hunk 1]
```diff
@@ -268,6 +269,48 @@ class ProvisionArtemisData(ArtemisGuestData, tmt.steps.provision.ProvisionStepDa
...
+                ]
+
+            if request_data is not None:
+                message_components += [
+                    '',
+                    dict_to_yaml(request_data)
+                    ]
+
+            message_components += [
+                f'Response: {response.status_code} {response.reason}',
```
Comment (psss) [inline tmt/steps/provision/artemis.py:302]: The `{response.status_code} {response.reason}` part seems to be a duplicating line 290. Is it intentional to show the status code and reason twice?

**Review by psss — COMMENTED** [review]
Looks good, just one question.

**Review by psss — APPROVED** [review]
Looks good, thanks!

File: tmt/steps/provision/artemis.py (hunk 1 above)
Comment (happz) [reply tmt/steps/provision/artemis.py:302]: Not necessary, I dropped the duplicities.

**Pull request merged**